driver/src/main/java/org/neo4j/driver/internal/BaseDriver.java

```diff
@@ -43,4 +54,68 @@ public boolean isEncrypted()
         return securityPlan.requiresEncryption();
     }
 
+    public <T> T transact( RetryLogic logic, AccessMode mode, Function<Transaction, T> work )
```

Comment:
It there any reason to put `RetryLogic` the first parameter? Otherwise I would suggest to put it the final parameter. As in C# this method would be more flexible if I give a default value
```
public <T> T transact(AccessMode mode, Function<Transaction, T> work, RetryLogic logic=null)
```
If a user do not supply a value when calling this method, then I could default to `contract.retryLogic`